src/layout/Grid/GridComponent.tsx

```diff
@@ -65,15 +69,29 @@ export function RenderGrid(props: PropsFromGenericComponent<'Grid'>) {
             labelSettings={labelSettings}
           />
         )}
-        {rowsInternal.map((row, rowIdx) => (
-          <GridRowRenderer
-            key={rowIdx}
-            row={row}
-            isNested={isNested}
-            mutableColumnSettings={columnSettings}
-            node={node}
-          />
-        ))}
+        <Table.Head>
+          {headerRows.map((row, rowIdx) => (
+            <GridRowRenderer
+              key={rowIdx}
+              row={row}
+              isNested={isNested}
+              mutableColumnSettings={columnSettings}
+              node={node}
+            />
+          ))}
+        </Table.Head>
+
+        <Table.Body>
+          {bodyRows.map((row, rowIdx) => (
+            <GridRowRenderer
+              key={rowIdx}
+              row={row}
+              isNested={isNested}
+              mutableColumnSettings={columnSettings}
+              node={node}
+            />
+          ))}
+        </Table.Body>
```

Comment:
This will break things. It's not always the case that header and body rows come in that order (header first, body last). These rows should be rendered out in the order they come, which sadly makes deduplicating the thead and tbody elements non-trivial.

Reply:
This exact change has been made and reverted before, we have to preserve the ordering. We have to essentialy group consecutive rows in a tbody, and close it when a header-row appears, so some more sophisticated logic is required

Reply:
Hmm så da må vi gjøre noe sånt her da?

```
import React from 'react';

function CustomTable({ rows }) {
  const sections = [];
  let currentSectionRows = [];
  let currentSectionType = null; // 'body' or 'header'

  rows.forEach((row, index) => {
    if (row.isHeader) {
      if (currentSectionType === 'body' && currentSectionRows.length > 0) {
        // Close the current tbody section
        sections.push(
          <tbody key={`tbody-${index}`}>
            {currentSectionRows.map((r, i) => (
              <tr key={i}>
                {r.cells.map((cell, j) => (
                  <td key={j}>{cell}</td>
                ))}
              </tr>
            ))}
          </tbody>
        );
        currentSectionRows = [];
      }

      if (currentSectionType === 'header' && currentSectionRows.length > 0) {
        // Close the current thead section
        sections.push(
          <thead key={`thead-${index}`}>
            {currentSectionRows.map((r, i) => (
              <tr key={i}>
                {r.cells.map((cell, j) => (
                  <th key={j}>{cell}</th>
                ))}
              </tr>
            ))}
          </thead>
        );
        currentSectionRows = [];
      }

      // Start a new header section
      currentSectionRows.push(row);
      currentSectionType = 'header';
    } else {
      if (currentSectionType === 'header' && currentSectionRows.length > 0) {
        // Close the current thead section
        sections.push(
          <thead key={`thead-${index}`}>
            {currentSectionRows.map((r, i) => (
              <tr key={i}>
                {r.cells.map((cell, j) => (
                  <th key={j}>{cell}</th>
                ))}
              </tr>
            ))}
          </thead>
        );
        currentSectionRows = [];
      }

      // Add to the current body section
      currentSectionRows.push(row);
      currentSectionType = 'body';
    }
  });

  // Add the last section
  if (currentSectionRows.length > 0) {
    if (currentSectionType === 'body') {
      sections.push(
        <tbody key="tbody-last">
          {currentSectionRows.map((r, i) => (
            <tr key={i}>
              {r.cells.map((cell, j) => (
                <td key={j}>{cell}</td>
              ))}
            </tr>
          ))}
        </tbody>
      );
    } else if (currentSectionType === 'header') {
      sections.push(
        <thead key="thead-last">
          {currentSectionRows.map((r, i) => (
            <tr key={i}>
              {r.cells.map((cell, j) => (
                <th key={j}>{cell}</th>
              ))}
            </tr>
          ))}
        </thead>
      );
    }
  }

  return <table>{sections}</table>;
}

export default CustomTable;

```

Reply:
Jepp, det kan funke. Potensielt mye duplisert kode der, men det er vel en slik fremgangsmåte man må bruke.

Reply:
Kun eksempel kode 👍